src/overmap.cpp

```diff
@@ -1557,6 +1578,11 @@ void overmap::generate(const overmap *north, const overmap *east,
                 roads_out.push_back(city(i.x, 0, 0));
             }
         }
+        for (auto &i : north->railroads_out) {
+            if (i.y == OMAPY - 1) {
+                railroads_out.push_back(city(i.x, 0, 0));
+            }
+        }
```

Comment:
Not strictly necessary here, but it would be much better if this crap was cleaned up before this PR lands.
4 directions, all with their own copypasted handling - ugly, error prone.

Reply:
Now that is just the copy of common roads mapgen - it certainly needs rework and railroads should be based upon reworked version.